vyper/semantics/types/subscriptable.py

```diff
@@ -84,10 +84,11 @@ def from_annotation(cls, node: vy_ast.Subscript) -> "HashMapT":
             )
 
         k_ast, v_ast = node.slice.elements
-        key_type = type_from_annotation(k_ast, DataLocation.STORAGE)
+        key_type = type_from_annotation(k_ast)
         if not key_type._as_hashmap_key:
             raise InvalidType("can only use primitive types as HashMap key!", k_ast)
 
+        # TODO: thread through actual location - might also be TRANSIENT
         value_type = type_from_annotation(v_ast, DataLocation.STORAGE)
```

Comment:
can we remove the STORAGE argument?

Reply:
i assumed we should keep it to enforce it's "storageable" - if smth had storage as an invalid location, then we should enforce it can't be used as value in maps